software/base/src/test/java/org/apache/brooklyn/entity/software/base/lifecycle/ScriptHelperIntegrationTest.java

```diff
@@ -0,0 +1,176 @@
+import org.apache.brooklyn.entity.software.base.AbstractSoftwareProcessSshDriver;
+import org.apache.brooklyn.entity.software.base.SoftwareProcessDriver;
+import org.apache.brooklyn.entity.software.base.VanillaSoftwareProcess;
+import org.apache.brooklyn.entity.software.base.VanillaSoftwareProcessImpl;
+import org.apache.brooklyn.location.ssh.SshMachineLocation;
+import org.apache.brooklyn.util.collections.MutableMap;
+import org.apache.brooklyn.util.core.task.system.internal.ExecWithLoggingHelpers;
+import org.apache.brooklyn.util.stream.ReaderInputStream;
+import org.apache.brooklyn.util.text.Strings;
+import org.apache.commons.io.output.ByteArrayOutputStream;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.io.StringReader;
+
+import static java.lang.String.format;
+
+public class ScriptHelperIntegrationTest extends BrooklynAppLiveTestSupport
+{
+    private static final Logger log = LoggerFactory.getLogger(ScriptHelperIntegrationTest.class);
+    
+    private Location loc;
+
+    @BeforeMethod(alwaysRun=true)
+    @SuppressWarnings("unchecked")
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        loc = app.getManagementContext().getLocationRegistry().getLocationManaged("localhost");
+    }
+
+    @Test(groups = "Integration")
+    public void testStopCommandWaitsToStopWithSigTerm() {
+        StopCommandSoftwareProcess entity = app.createAndManageChild(EntitySpec.create(StopCommandSoftwareProcess.class, StopCommandSoftwareProcessImpl.class));
+        entity.start(ImmutableList.of(loc));
+        VanillaSoftwareProcessSshDriver driver = (VanillaSoftwareProcessSshDriver) entity.getDriver();
+        String launchContents = Joiner.on('\n')
+                .join("#!/usr/bin/env bash",
+                        "function trap_handler_command {",
+                        "  echo stopping...",
+                        "  exit 13",
+                        "}",
+                        "trap \"trap_handler_command\" SIGTERM",
+                        "while true; do",
+                        "  sleep 1",
+                        "done");
+        driver.copyResource(ImmutableMap.of(), new ReaderInputStream(new StringReader(launchContents), "UTF-8"), "launch.sh", true);
+        driver.executeLaunchCommand("nohup bash launch.sh > /dev/null &");
+        ScriptHelper stopCommandScriptHelper = driver.stopCommandScriptHelper();
+        stopCommandScriptHelper.execute();
+        Assert.assertEquals(StringUtils.countMatches(stopCommandScriptHelper.getResultStdout(), "is still running"), 1, "SIGTERM should be tried one time");
+        Assert.assertFalse(stopCommandScriptHelper.getResultStdout().contains("stopped with SIGKILL"), "SIGKILL should not be sent after SIGTERM fails.");
```

Comment:
Are you saying that if `kill <pid>` returns non-zero, then it should not subsequently retry or do a `kill -9 <pid>`? Why not? It feels to me that we should escalate and do the `kill -9`.

Is it also worth asserting that the `stopCommandScriptHelper` exit code is correct?